### DISCUSSION TITLE
Accessing Claims using MSAL Authentication using Uno.Extensions

### DISCUSSION THREAD

**Discussion by arnold256** [unoplatform/uno.extensions#1966] [Q&A] [answered]
I'm integrating authentication into my application using Uno.Extensions and specifically the `AddMsal` method. Additionally, I'm using `IAuthenticationService` for the login process.

I'm interested in extracting specific information, like the username and other details, from the login token's claims. How can I access these claims?

**Comment by agneszitte** [comment, marked as answer]
cc @carldebilly, @nickrandolph

**Comment by nickrandolph** [comment]
@arnold256 the claims are stored in a token cache which you can access by adding ITokenCache as a dependency. There's an example of this in the testharness sample app [here](https://github.com/unoplatform/uno.extensions/blob/d7caf86801bd4d283f3eb5713a21e59c247adfc6/testing/TestHarness/TestHarness.Shared/Ext/Authentication/Msal/MsalAuthenticationHomeViewModel.cs#L46)
